internal/generator/pedersen/template/pedersen.go.tmpl

```diff
@@ -93,3 +99,74 @@ func (k *Key) VerifyKnowledgeProof(commitment {{.CurvePackage}}.G1Affine, knowle
     }
     return fmt.Errorf("proof rejected")
 }
+
+
+// Marshal
+
+func (pk *ProvingKey) WriteTo(w io.Writer) (int64, error) {
+	enc := curve.NewEncoder(w)
+	//l := len(pk.basis)
+	if err := enc.Encode(uint32(len(pk.basis))); err != nil {
+		return enc.BytesWritten(), err
+	}
+
+	for i := range pk.basis {
```

Comment:
can encode the slice directly (ie pass the slice as an argument to Encode) rather than a for loop on each element.

Same for decode.